proto/p4/GenericTable.md

```diff
@@ -0,0 +1,108 @@
+# GenericTable example p4info
+
+Below is an example for realizing MulticastGroupEntry as a GenericTable.
+The key (MatchFields) comprise of only the group_id. 2 different ways of
+defining the data fields has been presented. Only 1 of them is required.
+
+```
+generic_tables {
+  generic_table_type_id : 145
+  generic_table_type_name : "MulticastGroup"
+  preamble {
+    id: 45332650
+    name: "MulticastGroup"
+    alias: "multicast_group"
+  }
+  generic_match_fields {
+    id: 1
+    name: "multicast_group_id"
+    match_type: EXACT
+    type {
+      type : "bytes"
+      width : 32
+    }
+  }
+  union_refs {
+    id: 23557840
+  }
+  size: 1024
+}
+```
+
+In the below one, both `instance` and `port` are separate
+repeated fields. So the check on `len(instance_array) == len(port_array)`
+needs to be a runtime check. This however keeps implementation simpler
+and faster since we avoid further recursive nesting.
+
+`port` is a varbytes of max size 64 bits each. The field is repeated so it
+is defined as list of varbits through p4info.
+
+```
+unions {
+  preamble {
+    id: 23557840
+    name: "multicast_group_member_add"
+    alias: "multicast_group_member_add"
+  }
+  params {
+    id: 1
+    name: "instance"
+    repeated: true
```

Comment:
If you use a variable list type (GenericListTypeSpec), does that replace the "repeated" here? Is this "repeated" necessary?

Reply:
You are right `repeated` is redundant with List. I was thinking of having type as the below
```
    type {
      repetition_type : "list" // can be list, bag/multiset , set, unordered_set 
      type : "bytes"
      width : 32
    }
  }
```